ygen/protogen.go

```diff
@@ -387,6 +403,71 @@ func genProto3Msg(msg *yangDirectory, msgs map[string]*yangDirectory, state *gen
 	return append(msgDefs, msgDef), errs
 }
 
+// writeProtoEnums takes a map of enumerations, described as yangEnum structs, and returns
+// the mapped protobuf enum definition that is required. It skips any identified enumerated
+// type that is a simple enumerated leaf, as these are output as embedded enumerations within
+// each message. It returns a slice of srings containing the generated code, and the slice
+// of errors experienced.
+func writeProtoEnums(enums map[string]*yangEnum) ([]string, []error) {
+	var errs []error
+	var genEnums []string
+	for _, enum := range enums {
+		if enum.entry.Type.Kind == yang.Yenum && enum.entry.Type.Name == "enumeration" {
```

Comment:
Per comments elsewhere, define a helper function here (isSimpleEnumeration()?) and use that instead, here and elsewhere. It would be good to explain in that function why the name "enumeration" is significant, since it's not obvious. If that string is defined somewhere, it would be preferable to use the definition rather than the string literal.